packages/plugin-ext/src/plugin/text-editor.ts

```diff
@@ -334,6 +336,48 @@ export class TextEditorOptionsExt implements theia.TextEditorOptions {
         return undefined;
     }
 
+    get indentSize(): number {
+        return this._indentSize;
+    }
+
+    set indentSize(val: number | string | undefined) {
+        const indentSize = this.validateIndentSize(val);
+        if (!indentSize) {
+            return; // ignore invalid values
+        }
+
+        if (typeof indentSize === 'number') {
+            if (this._indentSize === indentSize) {
+                return;
+            }
+            this._indentSize = indentSize;
+        } else if (val === 'tabSize') {
+            this._indentSize = this._tabSize!;
```

Comment:
Suggestion: I believe this might yield incorrect values in case the `tabSize` is changed after `indentSize` has been set:

```ts
ext.tabSize = 2;
ext.indentSize = 'tabSize';
ext.tabSize = 4;
console.log(ext.indentSize) // prints '2'
```

It should store the `'tabSize'` value and return the current `tabSize` value from the `indentSize` getter.

Reply:
Indeed, this should be a better long term solution. Fixed in new version